### ISSUE TITLE
Mailpile loses pointers to previously downloaded mail (possibly suspend/resume related)

### ISSUE THREAD

**Issue by GregSutcliffe** [mailpile/Mailpile#2188]
Mailpile appears to have access to a bunch of previous mail. This *seems* to have occurred after a suspend/resume cycle while changing physical location. I checked mail *very* quickly before leaving home this morning (and I *think* it was fine then), but on arrival, mail between today and ~20th Nov is now unreadable.

The UI presents a traceback like this:
```
Message not found?

Failed to load and parse message data. Message may be corrupt!

Traceback (most recent call last):
  File "/usr/share/mailpile/mailpile/plugins/search.py", line 359, in _message
    msg = email.get_msg(pgpmime=False)
  File "/usr/share/mailpile/mailpile/mailutils/emails.py", line 930, in get_msg
    raise IndexError(_('Message not found?'))
IndexError: Message not found?

Message MID = 2BOE
Error = Message not found?
Details = Failed to load and parse message data. Message may be corrupt!
Data = []

If you think this is a bug, please file a report including the technical details above. 
```
After lengthy debugging in IRC with @BjarniRunar, it seems the metadata is intact in mailpile itself, but the "ptrs" field as been wiped somehow (seen via `hacks/metadata =2BOE`). The messages appear to be intact (tested via `cat /path/to/messagefile`) but Mailpile doesn't seem to know how to find it.

All other operations on the metadata seem fine (searching, threading of new messages, etc), so it seems that (a) something made Mailpile forget `ptrs` which needs to be understood, and (b) a way to fix `ptrs` (perhaps by rescanning the files in directory?) is needed.

**Comment by BjarniRunar** [comment]
Thank you very much for reporting this and helping me troubleshoot. I'm investigating this at my end.

**Comment by BjarniRunar** [comment]
I think I have fixed this in e5f01c6b18a8a225add1ec869b0ab5800c1f53e2, and I improved the CLI `rescan` command to make it easier to recover if your Mailpile hasn't already.

Basically, `rescan sources` should now fix this, assuming the messages were attached to a normal mail source (always the case if using the Mailpile web UI for configuring things).

Since a full `rescan sources` can take quite some time (press CTRL+C to abort, if you get bored), it is also possible to `rescan mailbox:NNNN` where NNNN is the ID of the mailbox that should contain the message, as identified by `print sys.mailbox`.

An example:

    mailpile> print sys.mailbox
    {
        "sys.mailbox": {        
            "0000": "/Mailpile$/mail",
            ...
            "0007": "src:9ee5c376331/INBOX",
            ...
        }
    }
    mailpile> rescan mailbox:0007
    ...

That should do the trick. Note that if you are running one of the `stable` .deb packages, you need to switch to the nightly releases to get the bugfix and a `rescan` command which handles this properly. Details here: https://www.mailpile.is/download/linux.html

I'm leaving this open for a few days to see if anyone has comments/complaints.
